app/controllers/media_controller.rb

```diff
@@ -228,21 +228,6 @@ def search
     results = search.results
     @total = search.total
 
-    # in the case of a search with tag_operator 'or', we
-    # execute two searches and merge the results, where media
-    # with the selected tags are now shown at the front of the list
-    if (search_params[:tag_operator] == "or") \
-      && (search_params[:all_tags] == "0") \
-      && (search_params[:fulltext].size >= 2)
-      params["search"]["all_tags"] = "1"
-      search_no_tags = Medium.search_by(search_params, params[:page])
-      search_no_tags.execute
-      results_no_tags = search_no_tags.results
-      results = (results + results_no_tags).uniq
-      @total = results.size
-      params["search"]["all_tags"] = "0"
-    end
```

Comment:
This was introduced in https://github.com/MaMpf-HD/mampf/commit/a67a2cf98b15e1b7b2acdb00ec0a182325e96c0b. Fabian wrote:

> updated media search by tags.
using the "or" operator now executes two searches,
one where the tags are included and one where
they are ignored. they are then combined so
that those with the selected tags are shown first
in the results.
also fixed the search with the "and" operator.

So that's the reason behind this.

Reply:
But I agree that this can be very misleading if the search *with* the tags did not yield any results, but the search *without* tags did. Then, the user only sees the results that were gathered *without* the specified tags and is really confused. That's why I agree that this part here can be removed.

Reply:
We could leave it in, but I really see no use case for that now (one and  half years later).

Reply:
Yeah, as this behavior might only lead to confusion I think we should get rid of it as you did 👍